File: 21-Subsets/solution.py
class Solution:
    def subsets(self, nums):
        n = 2 ** len(nums)

        res = []
        for counter in range(0, n):
            arr = []
            for j in range(0, len(nums)): 
                if (counter & (1 << j)) > 0: 
                    arr.append(nums[j])
            res.append(arr)

        return res

File: 21-Subsets/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        long long n = pow(2, nums.size());
        vector<vector<int>> res(n);
        vector<int> arr;

        for (long long i = 0; i < n; i++) {
            arr.clear();
            for (long long j = 0; j < nums.size(); j++) {
                if ((i & (1 << j)) > 0) arr.push_back(nums[j]);
            }
            res[i] = arr;
        }

        return res;
    }
};